apps/prairielearn/src/ee/lib/billing/plans.sql

```diff
@@ -1,24 +1,95 @@
--- BLOCK select_plan_grants_for_institution
+-- BLOCK select_plan_grants_for_context
 SELECT
-  *
+  pg.*
 FROM
-  plan_grants
+  plan_grants AS pg
 WHERE
-  institution_id = $institution_id
-  AND course_instance_id IS NULL
-  AND user_id IS NULL
-  and enrollment_id IS NULL;
+  -- Institution plan grant
+  (
+    $institution_id::bigint IS NOT NULL
+    AND $course_instance_id::bigint IS NULL
+    AND $enrollment_id::bigint IS NULL
+    AND $user_id::bigint IS NULL
+    AND institution_id = $institution_id
```

Comment:
Fwiw I think we don't need the `$institution_id::bigint IS NOT NULL` line because a NULL value in `$institution_id` will automatically make the comparison `institution_id = $institution_id` return NULL (which is falsey) for any value of `institution_id`. That being said, it also doesn't hurt to keep the more explicit checks above.

Reply:
Fair - and actually, backing up, I think this whole query can become much simpler:

```sql
-- BLOCK select_plan_grants_for_context
SELECT
  pg.*
FROM
  plan_grants AS pg
WHERE
  institution_id = $institution_id
  AND course_instance_id = $course_instance_id
  AND user_id = $user_id;
```

Reply:
Agreed!

Reply:
Done in b283082 - I re-discovered `IS NOT DISTINCT FROM` after wondering why `=` wasn't matching `NULL` columns 🤦